utils/tail/tail.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tail implements "tail -F" functionality following rotated logs
+package tail
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/exp/inotify"
+)
+
+type Tail struct {
+	reader   *bufio.Reader
+	filename string
+	file     *os.File
+	stop     chan bool
+}
+
+const retryReopenInterval = time.Second
+const maxReopenAttempts = 3
+
+// NewTail starts opens the given file and watches it for deletion/rotation
+func NewTail(filename string) (t *Tail, err error) {
+	t = &Tail{
+		filename: filename,
+	}
+	t.file, err = os.Open(t.filename)
+	if err != nil {
+		glog.Errorf("Open failed on %s", t.filename)
```

Comment:
nit: I prefer to only log where there error is actually captured, other wise you end up with the same error being logged multiple times. Instead, consider adding some context to the error (e.g. `fmt.Errorf("open failed on %s: %v", t.filename, err)`)